### PR TITLE
[Launcher]: Support ApplicationCompletionPolicy

### PR DESCRIPTION
**Pull request by yqwang-ms**

### CODE REVIEW

File: rest-server/src/models/job.js [hunk 1]
```diff
@@ -365,7 +365,7 @@ class Job {
   generateFrameworkDescription(data) {
     const gpuType = data.gpuType || null;
     const fancyRetryPolicy = (data.retryCount >= -1);
-    const killOnCompleted = (data.killAllOnCompletedTaskNumber > 0);
+    const minSucceededTaskCount = (data.killAllOnCompletedTaskNumber > 0) ? 1 : null;
```
Comment (abuccts) [inline rest-server/src/models/job.js]: `minSucceededTaskCount` = `data.killAllOnCompletedTaskNumber`?

Comment (yqwang-ms) [reply rest-server/src/models/job.js]: killAllOnAnyCompleted is equivalent to
"applicationCompletionPolicy": {
          "minFailedTaskCount": 1,
          "minSucceededTaskCount": 1
        },

---
In reply to: [196669188](https://github.com/Microsoft/pai/pull/742#discussion_r196669188) [](ancestors = 196669188)

Comment (yqwang-ms) [reply rest-server/src/models/job.js]: minSucceededTaskCount should be exactly equals to 1 to achieve the same behavior as killAllOnAnyCompleted 

---
In reply to: [196669668](https://github.com/Microsoft/pai/pull/742#discussion_r196669668) [](ancestors = 196669668,196669188)

Comment (abuccts) [reply rest-server/src/models/job.js]: I see. Should `minSucceededTaskCount` be `0` or `null` to disable `killAllOnAnyCompleted`?

Comment (yqwang-ms) [reply rest-server/src/models/job.js]: For how to refine RestServer later, please check user manual:

<tr>
      <td rowspan="2"><b>TensorFlow</td>
      <td>ParameterServer</td>
      <td><i>MinFailedTaskCount = 1<br>MinSucceededTaskCount = null</i></td>
      <td rowspan="2">Succeed a certain TaskRole is enough, and do not want to wait all Tasks to succeed:<br>Fail the Application immediately if any Task failed.<br>Succeed the Application immediately if Worker's all Tasks succeeded.</td>
    </tr>
    <tr>
      <td>Worker</td>
      <td><i>MinFailedTaskCount = 1</i><br>MinSucceededTaskCount = {Worker.TaskNumber}</td>
    </tr>


---
In reply to: [196669922](https://github.com/Microsoft/pai/pull/742#discussion_r196669922) [](ancestors = 196669922,196669668,196669188)

File: rest-server/src/models/job.js [hunk 2]
```diff
@@ -416,6 +414,10 @@ class Job {
             'diskMB': 0,
           },
         },
+        "applicationCompletionPolicy": {
+          "minFailedTaskCount": 1,
+          "minSucceededTaskCount": minSucceededTaskCount
```
Comment (abuccts) [inline rest-server/src/models/job.js]: Please fix [lint](https://travis-ci.org/Microsoft/pai/jobs/394413544): use single quotes, and add a trailing comma at the end.

File: rest-server/src/models/job.js (hunk 1 above)
Comment (yqwang-ms) [reply rest-server/src/models/job.js]: Should be null to disable killAllOnAnyCompleted

---
In reply to: [196670423](https://github.com/Microsoft/pai/pull/742#discussion_r196670423) [](ancestors = 196670423)

File: rest-server/src/models/job.js (hunk 2 above)
Comment (yqwang-ms) [reply rest-server/src/models/job.js]: ok

---
In reply to: [196670819](https://github.com/Microsoft/pai/pull/742#discussion_r196670819) [](ancestors = 196670819)

**Pull request merged**